addons/website_sale/views/templates.xml

```diff
@@ -1456,11 +1456,17 @@
             <t t-if="mode == ('new', 'billing') or (mode == ('edit', 'billing') and (can_edit_vat or 'vat' in checkout and checkout['vat']))">
                 <div t-attf-class="form-group #{error.get('company_name') and 'o_has_error' or ''} col-lg-6">
                     <label class="col-form-label font-weight-normal label-optional" for="company_name">Company Name</label>
-                    <input type="text" name="company_name" t-attf-class="form-control #{error.get('company_name') and 'is-invalid' or ''}" t-att-value="'commercial_company_name' in checkout and checkout['commercial_company_name'] or 'company_name' in checkout and checkout['company_name']" t-att-readonly="'1' if 'vat' in checkout and checkout['vat'] and not can_edit_vat else None" t-att-title="'Changing company name is not allowed once document(s) have been issued for your account. Please contact us directly for this operation.' if 'vat' in checkout and checkout['vat'] and not can_edit_vat else None" />
+                    <input type="text" name="company_name" t-attf-class="form-control #{error.get('company_name') and 'is-invalid' or ''}" t-att-value="'commercial_company_name' in checkout and checkout['commercial_company_name'] or 'company_name' in checkout and checkout['company_name']" t-att-readonly="'1' if 'vat' in checkout and checkout['vat'] and not can_edit_vat else None" />
+                    <t t-if="'vat' in checkout and checkout['vat'] and not can_edit_vat" >
+                        <small class="form-text text-muted">Changing company name is not allowed once document(s) have been issued for your account. Please contact us directly for this operation.</small>
+                    </t>
                 </div>
                 <div t-attf-class="form-group #{error.get('vat') and 'o_has_error' or ''} col-lg-6 div_vat">
                     <label class="col-form-label font-weight-normal label-optional" for="vat">TIN / VAT </label>
-                    <input type="text" name="vat" t-attf-class="form-control #{error.get('vat') and 'is-invalid' or ''}" t-att-value="'vat' in checkout and checkout['vat']" t-att-readonly="'1' if 'vat' in checkout and checkout['vat'] and not can_edit_vat else None" t-att-title="'Changing VAT number is not allowed once document(s) have been issued for your account. Please contact us directly for this operation.' if 'vat' in checkout and checkout['vat'] and not can_edit_vat else None" />
+                    <input type="text" name="vat" t-attf-class="form-control #{error.get('vat') and 'is-invalid' or ''}" t-att-value="'vat' in checkout and checkout['vat']" t-att-readonly="'1' if 'vat' in checkout and checkout['vat'] and not can_edit_vat else None"/>
+                    <t t-if="checkout['vat'] and not can_edit_vat">
```

Comment:
it is strange to don't check "'vat' in checkout" here... while we do it in 1466 no?

Reply:
Yes, I changed it because https://runbot.odoo.com/runbot/build/5196568 was telling me that `checkout` is now a `res.partner` instead of a `dict`.

However I can see some bug in the controller code. Just compare:

https://github.com/odoo/odoo/blob/37462d8a1175151cc07a921600d4f736af3acb80/addons/website_sale/controllers/main.py#L677
https://github.com/odoo/odoo/blob/37462d8a1175151cc07a921600d4f736af3acb80/addons/website_sale/controllers/main.py#L691

You should really consider start rolling out mypy over your codebase... :roll_eyes:

Reply:
It is true that not clean but i don’t see thé bug in controlleur...
While you do browserecord[] and dict[] ... maybe a read will be better... to avoid confusion

But i mean in you pr you didnt check if vat is present in checkout :) What will crash e.g. In checkout with a new shipping address 

Just an idea in my head :) a good tool too ! 🤔

Reply:
> But i mean in you pr you didnt check if vat is present in checkout :)

I was doing it in https://github.com/odoo/odoo/commit/c55ca35bc77faf3821a26e22feb4ed307ea0381e with:

```xml
<t t-if="checkout.get('vat') and not can_edit_vat">
```

But then [it failed](https://runbot.odoo.com/runbot/build/5196568) with: `AttributeError: 'res.partner' object has no attribute 'get'`

So, I thought: OK, this means `checkout` is no longer a `dict` but a `res.partner` instead. Then I don't need to check if `vat` exists *because it will always exist*. Then I updated the code to:

```xml
<t t-if="checkout['vat'] and not can_edit_vat">
```

But then [it failed](https://runbot.odoo.com/runbot/build/5199210) with: `KeyError: 'vat'`

So there you have the bug: The variable can be either a `res.partner` or a `dict`, depending on some scenarios that are outside of the control of the template code. `checkout` should have a stable type if you want to code the template sanely.

Reply:
I just fixed the bug with last commit.

Reply:
in stable you could just keep `'vat' in checkout` that work with browse record and dict ...

Reply:
We're not in stable.

Reply:
yes, you are right, but btw this change will not fix your problem.
Browse record or dict, you need to check if 'vat' is in checkout ...

Reply:
Yikes true. Fixed now.